File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.scala
object Solution {
    def reverseWords(s: String): String = {
      var ans = ""
      var temp = ""

      for (i <- s) {
        if (i == ' ') {
          ans += temp.reverse
          ans += " "
          temp = ""
        } else {
          temp += i
        }
      }

      ans + temp.reverse
    }
}

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.ts
function reverseWords(s: string): string {
  let ans = "";
  let temp = "";

  for (let i = 0; i < s.length; i++) {
    if (s[i] === " ") {
      ans += temp.split("").reverse().join("");
      ans += " ";
      temp = "";
    } else {
      temp += s[i];
    }
  }

  return ans + temp.split("").reverse().join("");
};